pkg/apis/servicecatalog/types.go

```diff
@@ -56,15 +56,44 @@ type BrokerSpec struct {
 // BrokerAuthInfo is a union type that contains information on one of the authentication methods
 // the the service catalog and brokers may support, according to the OpenServiceBroker API
 // specification (https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md).
-//
-// Note that we currently restrict a single broker to have only one of these fields
-// set on it.
 type BrokerAuthInfo struct {
-	// BasicAuthSecret is a reference to a Secret containing auth information the
-	// catalog should use to authenticate to this Broker using basic auth.
-	BasicAuthSecret *v1.ObjectReference
+	// Basic provides configuration for basic authentication
+	Basic *BasicAuthConfig
+	// Bearer provides configuration for bearer token authentication
+	Bearer *BearerTokenAuthConfig
 }
 
+// BasicAuthConfig provides config for the basic authentication.
+type BasicAuthConfig struct {
+	// SecretRef is a reference to a Secret containing information the
+	// catalog should use to authenticate to this Broker.
+	//
+	// Required at least one of the fields:
+	// - Secret.Data["username"] - username used for authentication
+	// - Secret.Data["password"] - password or token needed for authentication
+	SecretRef *v1.ObjectReference
+}
+
+// BearerTokenAuthConfig provides config for the bearer token authentication.
```

Comment:
I think we should be clear about exactly what the behavior this has.  I'd say the following:

```go
// BearerTokenAuthConfig provides configuration to send an opaque value as a bearer token.
// The value is referenced from the 'token' field of the given secret.  This value should only
// contain the token value and not the `Bearer` scheme.
```